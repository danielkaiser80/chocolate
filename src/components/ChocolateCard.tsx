import {
  Box,
  Card,
  CardContent,
  Link as MUILink,
  Typography,
} from "@mui/material";
import { Link } from "wouter";

interface PriceInfo {
  cheapestPricePer100g: number;
  cheapestShopLink: string;
  averagePricePer100g: number;
}
interface ChocolateCardProps {
  name: string;
  brand: string;
  priceInfo?: PriceInfo;
  id: string;
}

const NUMBER_FORMAT = {
  maximumFractionDigits: 2,
};
const ChocolateCard = ({ name, brand, priceInfo, id }: ChocolateCardProps) => (
  <Card>
    <CardContent>
      <Box mb={2}>
        <Typography variant="h5">üç´ {name}</Typography>
        <Typography variant="body1">Hersteller: {brand}</Typography>
      </Box>

      {priceInfo ? (
        <>
          <Box mb={2}>
            <Typography variant="body2">
              G√ºnstigster Preis:{" "}
              {priceInfo.cheapestPricePer100g.toLocaleString(
                "de",
                NUMBER_FORMAT,
              )}{" "}
              ‚Ç¨ je 100g
            </Typography>
            <MUILink href={priceInfo.cheapestShopLink}>Jetzt kaufen</MUILink>
          </Box>
          <Box>
            <Typography variant="body2">
              Durchschnittspreis:{" "}
              {priceInfo.averagePricePer100g.toLocaleString(
                "de",
                NUMBER_FORMAT,
              )}{" "}
              ‚Ç¨ je 100g
            </Typography>
          </Box>
        </>
      ) : (
        <Box>
          <Typography variant="body2">Derzeit nicht verf√ºgbar üò≠</Typography>
        </Box>
      )}

      <hr />

      <Link href={`/chocolates/${id}`}>√ñffnen der Detail-Seite</Link>
    </CardContent>
  </Card>
);

export default ChocolateCard;
