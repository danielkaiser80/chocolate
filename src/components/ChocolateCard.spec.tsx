import ChocolateCard from "./ChocolateCard.tsx";
import { render } from "@testing-library/react";
import { describe } from "vitest";

describe(ChocolateCard, () => {
  it("should render non available chocolate", () => {
    const result = render(
      <ChocolateCard brand={"Ritter Sport"} name={"Alpenmilch"} />,
    );
    expect(result).toMatchSnapshot();
  });

  it("should render available chocolate", () => {
    const result = render(
      <ChocolateCard
        brand={"Ritter Sport"}
        name={"Alpenmilch"}
        priceInfo={{
          averagePricePer100g: 1.01,
          cheapestShopLink: "http://example.com/test",
          cheapestPricePer100g: 0.99,
        }}
      />,
    );
    expect(result).toMatchSnapshot();
  });
});
