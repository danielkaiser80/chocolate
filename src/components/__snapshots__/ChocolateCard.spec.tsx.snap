// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChocolateCard > should render available chocolate 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              üç´ 
              Alpenmilch
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Hersteller: 
              Ritter Sport
            </p>
          </div>
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              G√ºnstigster Preis:
               
              0,99
               
              ‚Ç¨ je 100g
            </p>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
              href="http://example.com/test"
            >
              Jetzt kaufen
            </a>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Durchschnittspreis:
               
              1,01
               
              ‚Ç¨ je 100g
            </p>
          </div>
          <hr />
          <a
            href="/chocolates/2"
          >
            √ñffnen der Detail-Seite
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            üç´ 
            Alpenmilch
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Hersteller: 
            Ritter Sport
          </p>
        </div>
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            G√ºnstigster Preis:
             
            0,99
             
            ‚Ç¨ je 100g
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
            href="http://example.com/test"
          >
            Jetzt kaufen
          </a>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Durchschnittspreis:
             
            1,01
             
            ‚Ç¨ je 100g
          </p>
        </div>
        <hr />
        <a
          href="/chocolates/2"
        >
          √ñffnen der Detail-Seite
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChocolateCard > should render non available chocolate 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              üç´ 
              Alpenmilch
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Hersteller: 
              Ritter Sport
            </p>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Derzeit nicht verf√ºgbar üò≠
            </p>
          </div>
          <hr />
          <a
            href="/chocolates/1"
          >
            √ñffnen der Detail-Seite
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            üç´ 
            Alpenmilch
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Hersteller: 
            Ritter Sport
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Derzeit nicht verf√ºgbar üò≠
          </p>
        </div>
        <hr />
        <a
          href="/chocolates/1"
        >
          √ñffnen der Detail-Seite
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
